src/Compilers/CSharp/Portable/Lowering/ExtensionMethodReferenceRewriter.cs

```diff
@@ -149,7 +149,7 @@ static RefKind argumentRefKindFromReceiverRefKind(RefKind receiverRefKind)
         [return: NotNullIfNotNull(nameof(method))]
         private static MethodSymbol? VisitMethodSymbolWithExtensionRewrite(BoundTreeRewriter rewriter, MethodSymbol? method)
         {
-            if (method is { OriginalDefinition.ContainingSymbol: NamedTypeSymbol { IsExtension: true } declaringTypeDefinition } &&
+            if (method?.GetIsNewExtensionMember() == true &&
```

Comment:
Previously the assert here and below was on `method.OriginalDefinition` and now it is on just `method`, is that change intended?

Reply:
> is that change intended?

That should not affect the result.